packages/app/src/settings/SettingsCard.vue

```diff
@@ -43,20 +45,87 @@
 
 <script lang="ts" setup>
 import type { FunctionalComponent, SVGAttributes } from 'vue'
+import { ref, ComponentPublicInstance, computed, watchEffect, nextTick } from 'vue'
 import Collapsible from '@cy/components/Collapsible.vue'
 import ListRowHeader from '@cy/components/ListRowHeader.vue'
+import { useRoute } from 'vue-router'
 
-defineProps<{
+const props = defineProps<{
   title: string
+  name?: string
   description: string
   icon: FunctionalComponent<SVGAttributes, {}>
   maxHeight: string
 }>()
 
-</script>
+const route = useRoute()
+
+const initiallyOpen = computed(() => {
+  if (!props.name || !route.query.section) {
+    return false
+  }
+
+  return route.query.section === props.name
+})
+
+const root = ref<ComponentPublicInstance>()
+
+/**
+ * This feature is used for opening and scrolling to a specific
+ * setting, often as a result of doing something else
+ * somewhere in the App.
+ *
+ * One example use case is enabling Desktop Notifications.
+ * When a user selects "Enable desktop notifications" on the
+ * Specs page, the desired behavior is:
+ *
+ * 1. Redirect to the /settings page
+ * 2. Open the relevant settings card (Device Settings, in this case)
+ * 3. Scroll to the #notifications anchor
+ *
+ * Example usage:
+ *
+ *
+ * router.push({
+ *   path: '/settings',         // page to redirect to
+ *   query: {
+ *     section: 'device',       // section to open by default
+ *     anchor: 'notifications'  // anchor to scroll to
+ *   }
+ * })
+ *
+ * @see https://github.com/cypress-io/cypress/issues/27090
+ */
+function maybeScrollToAnchor () {
+  if (!initiallyOpen.value) {
+    return
+  }
 
-<style lang="scss" scoped>
-.settings-card-header {
-  grid-template-columns: auto auto 1fr auto;
+  if (!route.query.anchor) {
+    // Do nothing - no anchor query parameter.
```

Comment:
What's the purpose of this if? Should we log something or return?

Reply:
Yes we should - looks like this was not addressed, I'll push it in the main feature branch.